save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/demo/DemoView.kt

```diff
@@ -0,0 +1,79 @@
+/**
+ * View with demo for diktat and ktlint
+ */
+
+package com.saveourtool.save.frontend.components.views.demo
+
+import com.saveourtool.save.demo.DemoInfo
+import com.saveourtool.save.demo.DemoRunRequest
+import com.saveourtool.save.frontend.components.basic.cardComponent
+import com.saveourtool.save.frontend.components.basic.demoRunComponent
+import com.saveourtool.save.frontend.externals.reactace.AceThemes
+import com.saveourtool.save.frontend.utils.*
+import com.saveourtool.save.utils.Languages
+
+import csstype.*
+import js.core.get
+import react.*
+import react.dom.html.ReactHTML.div
+import react.router.useParams
+
+private val backgroundCard = cardComponent(hasBg = true, isPaddingBottomNull = true)
+
+val demoView = demoView()
+
+private fun demoView(): VFC = VFC {
+    useBackground(Style.WHITE)
+    val (selectedTheme, setSelectedTheme) = useState(AceThemes.preferredTheme)
+    val params = useParams()
```

Comment:
Why do we need to use `useParams` with `VFC` instead of passing params from `react-router` into `FC` (as usual)? Is it because of diktat demo "legacy" support?

Reply:
nope, `BasicRouting.kt` setups routing from url to some view. And view paths (as in Spring) are parsed by `useParams`.
It's the same as `withRouter` works inside or `com.saveourtool.save.frontend.components.views.TestExecutionDetailsViewKt#testExecutionDetailsView`

Reply:
oh ok, but it seems that it is better to use unified high(er)-level params parsing logic where applicable